<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="0">
     <widget class="QFrame" name="frame">
      <property name="minimumSize">
       <size>
        <width>394</width>
        <height>190</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>390</width>
        <height>190</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QFrame" name="frameEdit">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>130</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>400</width>
           <height>600</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="2">
           <widget class="QLabel" name="labelFrom">
            <property name="text">
             <string>From</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="aEditFill">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelB">
            <property name="text">
             <string>B=</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelA">
            <property name="text">
             <string>A=</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="bEditFill">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QLabel" name="labelStep">
            <property name="text">
             <string>Step</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="labelC">
            <property name="text">
             <string>C=</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="cEditFill">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="labelTo">
            <property name="text">
             <string>To</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLineEdit" name="fromEditFill">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLineEdit" name="toEditFill">
            <property name="text">
             <string>1</string>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QLineEdit" name="stepEditFill">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="4">
           <widget class="QComboBox" name="comboBox">
            <property name="insertPolicy">
             <enum>QComboBox::InsertAtBottom</enum>
            </property>
            <item>
             <property name="text">
              <string>f(x) = A*(x*x) + B*x + C</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>f(x) = A * sin(x) + B * cos( C*x )</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>f(x) = A*log( B*x )</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>f(x) = A / ( sin(x*x) * B )</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_calculate"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayoutButton">
         <item>
          <widget class="QPushButton" name="startB">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pauseB">
           <property name="text">
            <string>Pause</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="breakB">
           <property name="text">
            <string>Break</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1" rowspan="4">
     <widget class="Graphics" name="widget" native="true"/>
    </item>
    <item row="3" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Graphics</class>
   <extends>QWidget</extends>
   <header>graphics.h</header>
   <container>1</container>
   <slots>
    <slot>addPoint(QPointF)</slot>
    <slot>SetPointer(QVector*)</slot>
    <slot>setPermit(bool)</slot>
    <slot>setNew(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>stepEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedStep(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedSwitchFunc(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fromEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedFrom(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>379</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedC(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedA(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>149</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startB</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotEnabled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedTo(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>sendPoint(QPointF)</signal>
   <receiver>widget</receiver>
   <slot>addPoint(QPointF)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseB</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotPause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>breakB</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slotBreak()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>353</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>sendPermit(bool)</signal>
   <receiver>widget</receiver>
   <slot>setPermit(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>sendNew(bool)</signal>
   <receiver>widget</receiver>
   <slot>setNew(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>405</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bEditFill</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>valueChangedB(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>sendPoint(QPointF)</signal>
  <signal>sendZero(int)</signal>
  <signal>sendVectorPointer(QVector*)</signal>
  <signal>sendPermit(bool)</signal>
  <signal>sendNew(bool)</signal>
  <slot>valueChangedA(QString)</slot>
  <slot>valueChangedB(QString)</slot>
  <slot>valueChangedC(QString)</slot>
  <slot>valueChangedFrom(QString)</slot>
  <slot>valueChangedTo(QString)</slot>
  <slot>valueChangedStep(QString)</slot>
  <slot>valueChangedSwitchFunc(int)</slot>
  <slot>slotEnabled()</slot>
  <slot>slotBreak()</slot>
  <slot>slotPause()</slot>
  <slot>slot1()</slot>
 </slots>
</ui>
